spec/services/markdown_processor/parser_spec.rb

```diff
@@ -40,6 +40,29 @@ def generate_and_parse_markdown(raw_markdown)
     expect(number_of_triple_backticks).to eq(0)
   end
 
+  it "does not insert extra newlines in codeblocks when using tildes" do
+    code_block = "~~~\nhello\n   there\n~~~"
+    parsed = generate_and_parse_markdown(code_block)
+    rendered_code_block = parsed.scan(%r{<code>.*?</code>}m).first
```

Comment:
I'm pretty sure this regex is wrong in a way that doesn't matter for this application - but that sets a bad precedent for someone revisiting this later.

Reply:
We could ditch the regex and use Nokogiri instead:

```ruby
    rendered_code_block = Nokogiri::HTML(parsed).xpath("//code")
    expect(rendered_code_block.to_s).to eq("<code>hello\n   there\n</code>")
```

Reply:
this is why I bring attention to my shortcomings - thank you! 

It will be more important when I move on to the case where there are two code blocks handling #13823 to have something better than my greedy regex.